File: frontend/public/examples/cpp/objects.cpp
// Creating objects in C++

#include <iostream>
#include <string>

class Person {
private:
    std::string name;
    int age;

public:
    Person(const std::string& name, int age) : name(name), age(age) {}

    std::string greet() const {
        return "Hello, my name is " + name + " and I am " + std::to_string(age) + " years old.";
    }
};

int main() {
    Person person("Alice", 30);
    std::cout << person.greet() << std::endl;
    return 0;
}

File: frontend/public/examples/rust/objects.rs
// Creating objects in Rust

struct Person {
    name: String,
    age: u32,
}

impl Person {
    fn new(name: &str, age: u32) -> Self {
        Person {
            name: name.to_string(),
            age,
        }
    }

    fn greet(&self) -> String {
        format!("Hello, my name is {} and I am {} years old.", self.name, self.age)
    }
}

fn main() {
    let person = Person::new("Alice", 30);
    println!("{}", person.greet());
}
